### PR TITLE
Remove duplicates filters and allow sync post termination

### PR DESCRIPTION
**Pull request by tlovett1**
Remove duplicate `ep_post_sync_args` filter and add `ep_prepare_post_args` filter. If ep_post_sync_args is empty, don't sync

### CODE REVIEW

**Comment by tlovett1** [comment]
@AaronHolbrook #reviewmerge

File: classes/class-ep-api.php [hunk 1]
```diff
@@ -608,7 +608,7 @@ public function prepare_post( $post_id ) {
 			//'site_id'         => get_current_blog_id(),
 		);
 
-		$post_args = apply_filters( 'ep_post_sync_args', $post_args, $post_id );
+		$post_args = apply_filters( 'ep_prepare_post_args', $post_args, $post_id );
```
Comment (AaronHolbrook) [inline classes/class-ep-api.php]: @tlovett1 changing this here could result in buggy behavior while performing a bulk index (this function is called from two places, sync_post and queue_post)

Comment (AaronHolbrook) [inline classes/class-ep-api.php]: called here: https://github.com/10up/ElasticPress/blob/bugfix/duplicate-filter/classes/class-ep-sync-manager.php#L108

and here: https://github.com/10up/ElasticPress/blob/bugfix/duplicate-filter/bin/wp-cli.php#L342

**Comment by tlovett1** [comment]
@AaronHolbrook I am merging this. The build error was due to a Travis problem.

**Pull request merged**